src/Xamarin.Android.Build.Tasks/Tasks/CreateNativeLibraryArchive.cs

```diff
@@ -61,13 +61,20 @@ public override bool Execute ()
 					outDirInfo.CreateSubdirectory (abi);
 				MonoAndroidHelper.CopyIfChanged (lib.ItemSpec, Path.Combine (OutputDirectory, abi, Path.GetFileName (lib.ItemSpec)));
 			}
-				
+
 			// Archive native libraries in a zip.
-			var zip = new ZipFile (new System.Text.UTF8Encoding (false));
-			zip.AddDirectory (OutputDirectory, outDirInfo.Name);
-			string outpath = Path.Combine (outDirInfo.Parent.FullName, "__AndroidNativeLibraries__.zip");
-			if (Files.ArchiveZip (outpath, f => zip.Save (f)))
-				Log.LogDebugMessage ("Saving contents to " + outpath);
+			using (var stream = new MemoryStream ()) {
+				using (var zip = new ZipArchive (stream, ZipArchiveMode.Create, true, new System.Text.UTF8Encoding (false))) {
```

Comment:
When you have `using` after `using`, you don't need to separately indent the 2nd (3rd, 4th...) `using`:

    using (var a = ...)
    using (var b = ...)
    using (var c = ...) {
        ...
    }

This helps get code too deeply nested. Thus:

    using (var stream = new MemoryStream ())
    using (var zip = new ZipArchive (...)) {
        ...
    }